### ISSUE TITLE
Unable to inject params via constructor in Workers after upgrading to 3.4.0

### ISSUE THREAD

**Issue by ricardomgarciaf** [InsertKoinIO/koin#1554]
**Describe the bug**
Before Koin 3.4.0, I was able to inject any parameters directly in the Worker constructor. 
This used to work in previous versions:

```
worker {
        TestWorker(
            context = androidContext(),
            params = get(),
            testUseCase = get()
        )
}
```
Now I'm getting following exception when trying to launch the worker.

```
java.lang.NoSuchMethodException: com.example.myapplication.TestWorker.<init> [class android.content.Context, class androidx.work.WorkerParameters]
at java.lang.Class.getConstructor0(Class.java:2363)
at java.lang.Class.getDeclaredConstructor(Class.java:2201)
at androidx.work.WorkerFactory.createWorkerWithDefaultFallback(WorkerFactory.java:95)
at androidx.work.impl.WorkerWrapper.runWorker(WorkerWrapper.java:243)
at androidx.work.impl.WorkerWrapper.run(WorkerWrapper.java:145)
at androidx.work.impl.utils.SerialExecutorImpl$Task.run(SerialExecutorImpl.java:96)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
at java.lang.Thread.run(Thread.java:1012)
2023-04-03 17:26:29.207  5744-6415  WM-WorkerWrapper        com.example.myapplication            E  Could not create Worker com.example.myapplication.TestWorker
```

The exception goes away by removing `testUseCase` from the contructor params and injecting it as a field in the Worker.

I didn't find anywhere in the documentation where this change was listed, which makes me think this is an issue.

**To Reproduce**
Declare a worker in the injection graph, injecting any other param besides context and params.

**Expected behavior**
Constructor injection should work in workers.

**Koin project used and used version (please complete the following information):**
```
"io.insert-koin:koin-core:3.4.0"
"io.insert-koin:koin-androidx-workmanager:3.4.0"
```

**Comment by tatocaster** [comment]
This change is breaking the work manager part:
[https://github.com/InsertKoinIO/koin/commit/58d8cab448cc72eca1398436ae332bd3b468715d](https://github.com/InsertKoinIO/koin/commit/58d8cab448cc72eca1398436ae332bd3b468715d)

if you pass the custom qualifier in
```
worker(named<TestWorker>()){
```
it will work.

cc : @arnaudgiuliani

**Comment by ricardomgarciaf** [comment]
@tatocaster That did the trick. Thank you!

**Comment by Pitel** [comment]
Got the same issue.

Naming the definition also solved it for me:
```kotlin
workerOf(::MyWorker) { named<MyWorker>() }
```

**Comment by Nek-12** [comment]
Same story here. Migrated to field injection because of this.

**Comment by arnaudgiuliani** [comment]
Yep, sorry a regression for due to some Koin Annotations API. I'm on it 👍

**Comment by arnaudgiuliani** [comment]
Fixed in `3aab37c7c69d8674e02ff4563ef599535c1654a8` - see next patch for `koin-android:3.4.1`
